doc/using_the_march_exoskeleton/error_codes.rst

```diff
@@ -25,10 +25,117 @@ without having to repeat the whole error description.
 
 .. _e100:
 
-``E100``: Placeholder error
----------------------------
-This should be a short description of the error.
+``E100``: Invalid actuation mode
+--------------------------------
+An invalid actuation mode was used to perform an action.
 
-**Cause:** You did this thing wrong
+**Cause:** The motor controllers have two modes for actuation: position and effort.
+With position you can only actuate to a target position using ``actuateRad()``.
+In effort mode you can only actuate using a target torque using ``actuateTorque()``.
+This error was caused by mixing the actuate methods and actuation modes. See
+the detailed description for which mode the controller was actually in.
 
-**Fix:** Do this and that and then it may be solved
+**Fix:** Check if the actuation mode defined in the :hardware-interface:`robots yaml <march_hardware_builder/robots>`
+you use is consistent with the methods you are using to actuate the joint.
+
+
+.. _e101:
+
+``E101``: Invalid actuate position
+----------------------------------
+Invalid IU position command.
+
+**Cause:** The position you are trying to actuate to is not a valid target within the range of the joint.
+
+**Fix:** Check that the IU limits defined in the :hardware-interface:`robots yaml <march_hardware_builder/robots>`
+under ``joints/<joint_name>/imotioncube/encoder`` match the actual limits of the joint.
+
+
+.. _e102:
+
+``E102``: Encoder reset
+-----------------------
+An encoder has reset and reads an incorrect value.
+
+**Cause:** The cause is still unknown.
+
+**Fix:** This error happens sometimes during startup and the only thing you
+can do is try again. This error will probably be removed once this issue has
+been fixed.
+
+
+.. _e103:
+
+``E103``: Outside hard limits
+-----------------------------
+A joint is outside its defined hard limits.
+
+**Cause:** A joint was found to be outside its defined hard limits in the
+:hardware-interface:`robots yaml <march_hardware_builder/robots>`.
+
+**Fix:** Check that the IU limits defined in the :hardware-interface:`robots yaml <march_hardware_builder/robots>`
+under ``joints/<joint_name>/imotioncube/encoder`` match the actual limits of the joint and recalibrate if necessary.
```

Comment:
maybe also as a fix move the exoskeleton to a logical position (within limits)?

Reply:
True, however this should not happen since you can not physically be outside of hard limits. So this is something that should be fixed

Reply:
Right, True. But if this error occurs it would be false values? Because if the joints are indeed outside hard limits the fix would be a bit more complicated. False values could be fixed by moving the index of the encoder a bit so it would read a new position.